davidl-kinley/server.js

```diff
@@ -0,0 +1,193 @@
+  CLIENT.query(
+    `UPDATE articles
+    SET
+      title=$1, author=$2, "authorUrl"=$3, category=$4, "publishedOn"=$5, body=$6
+    WHERE article_id=$7;
+    `,
+    [
+      request.body.title,
+      request.body.author,
+      request.body.authorUrl,
+      request.body.category,
+      request.body.publishedOn,
+      request.body.body,
+      request.params.id
+    ]
+  )
+  .then(function() {
+    response.send('update complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+APP.delete('/articles/:id', function(request, response) {
+  // COMMENTed: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // The following code pertains to number 3 on the diagram and it also sends a result back to the browser in a response which is numbers 4 and 5. This is interacting with the deleteRecord method. This is the D in CRUD.
+  CLIENT.query(
+    `DELETE FROM articles WHERE article_id=$1;`,
+    [request.params.id]
+  )
+  .then(function() {
+    response.send('Delete complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+APP.delete('/articles', function(request, response) {
+  // COMMENTed: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // The following code corresponds to numbers 3, 4 and 5 on the diagram. This is interacting with the truncateTable method on article.js. This is also the D in CRUD.
+  CLIENT.query(
+    'DELETE FROM articles;'
+  )
+  .then(function() {
+    response.send('Delete complete')
+  })
+  .catch(function(err) {
+    console.error(err);
+  });
+});
+
+// COMMENTed: What is this function invocation doing?
+// If there isnt already a table setup for the articles, it is setting up a table with the propery column and attributes. Then it runs the function loadArticles.
+loadDB();
+
+APP.listen(PORT, function() {
+  console.log(`Server started on port ${PORT}!`);
+});
+
+
+//////// ** DATABASE LOADER ** ////////
+////////////////////////////////////////
+function loadArticles() {
+  // COMMENTed: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // The following code all happens in the DB and does not pertain to a number on the diagram. It doesnt interact with any method in the article.js file. This is the U in CRUD.
```

Comment:
Technically, this isn't CRUD in the sense of a `CRUD route`. But you can still consider it one of the four, just not `U`. Go over the function again.